arroyo-df/src/plan_graph.rs

```diff
@@ -61,10 +77,11 @@ pub(crate) async fn get_arrow_program(
             .local_logical_plan_graph
             .node_weight(node_index)
             .unwrap();
-        match logical_extension {
+
+        let new_node = match logical_extension {
             crate::LogicalPlanExtension::TableScan(logical_plan) => {
                 let LogicalPlan::TableScan(table_scan) = logical_plan else {
-                    bail!("expected table scan")
+                    panic!("expected table scan")
```

Comment:
If we have a Result return, I generally perefer bailing. I'm guessing this was to better identify where the error was happening?

Reply:
In my mind, they serve two different purposes and it's useful to have both. Returning a result makes sense when it's a user-caused error that we want to report back to them. But when it's a programming error (there should be no way to end up with a LogicalPlanExtension::TableScan that doesn't contain a TableScan) there's no good reason to return that error message to the user; panicking is a sign to us that there's a bug we need to fix.

Ideally stuff like this would be encoded in the type system though

Reply:
Gotchya, I like `unreachable!()` to indicate that we don't believe it is possible to get there.